core-tests/shared/src/test/scala/zio/stm/ZSTMSpec.scala

```diff
@@ -463,9 +466,38 @@ object STMSpec extends ZIOBaseSpec {
 
         assertM(chain(10000).run, fails(equalTo(10000)))
       }
+    ),
+    suite("STM environment")(
+      testM("access environment and provide it outside transaction") {
+        STMEnv.make(0).flatMap { env =>
+          ZSTM.accessSTM[STMEnv](_.ref.update(_ + 1)).commit.provide(env) *>
```

Comment:
I would keep the naming, i.e. `accessM` rather than `accessSTM`.

Reply:
There was a similar discussion here https://github.com/zio/zio/pull/2337#discussion_r351704196

Reply:
I don't think there's consistency in the code yet. One could argue for `accessM` (look at `mapM` in STM).

Reply:
I'd say let's keep the M suffix everywhere. While it might not be ideal, it is somewhat accepted convention, especially if you look at Haskell counterparts.